packages/web5/src/record.ts

```diff
@@ -0,0 +1,245 @@
+  //   // a top-level property to `web5.dwn.records.write()`.
+  //   let data;
+  //   if (options.data !== undefined) {
+  //     delete updateMessage.dataCid;
+  //     delete updateMessage.dataSize;
+  //     data = options.data;
+  //     delete options.data;
+  //   }
+
+  //   // Throw an error if an attempt is made to modify immutable properties. `data` has already been handled.
+  //   const mutableDescriptorProperties = ['dataCid', 'dataSize', 'dateModified', 'datePublished', 'published'];
+  //   Record.#verifyPermittedMutation(Object.keys(options), mutableDescriptorProperties);
+
+  //   // If a new `dateModified` was not provided, remove it from the updateMessage to let the DWN SDK auto-fill.
+  //   // This is necessary because otherwise DWN SDK throws an Error 409 Conflict due to attempting to overwrite a record
+  //   // when the `dateModified` timestamps are identical.
+  //   if (options.dateModified === undefined) {
+  //     delete updateMessage.dateModified;
+  //   }
+
+  //   // If `published` is set to false, ensure that `datePublished` is undefined. Otherwise, DWN SDK's schema validation
+  //   // will throw an error if `published` is false but `datePublished` is set.
+  //   if (options.published === false && updateMessage.datePublished !== undefined) {
+  //     delete updateMessage.datePublished;
+  //   }
+
+  //   // Set the record ID and context ID, if any.
+  //   updateMessage.recordId = this.#recordId;
+  //   updateMessage.contextId = this.#contextId;
+
+  //   // Attempt to write the changes to mutable properties to the DWN.
+  //   const { message = null, record = null, status } = await this.#dwn.records.write(this.#target, {
+  //     author  : this.#author,
+  //     data,
+  //     message : {
+  //       ...updateMessage,
+  //     },
+  //   });
+
+  //   if (status.code === 202 && record) {
+  //     // Only update the local Record instance mutable properties if the record was successfully (over)written.
+  //     mutableDescriptorProperties.forEach(property => {
+  //       this.#descriptor[property] = record[property];
+  //     });
+  //     // Only update data if `dataSize` is less than DWN 'max data size allowed to be encoded'.
+  //     if (data !== undefined) {
+  //       this.#readableStream = (record.dataSize <= DwnConstant.maxDataSizeAllowedToBeEncoded) ? record.data : null;
+  //       this.#encodedData = null; // Clear `encodedData` in case it was previously set.
+  //     }
+  //   }
+
+  //   return { message, status };
+  // }
+
+  /**
+   * Utility methods.
+   */
+
+  /**
+   * Called by `JSON.stringify(...)` automatically.
+   */
+  toJSON() {
+    return {
+      author        : this.author,
+      target        : this.target,
+      recordId      : this.id,
+      contextId     : this.contextId,
+      dataFormat    : this.dataFormat,
+      dateCreated   : this.dateCreated,
+      interface     : this.interface,
+      method        : this.method,
```

Comment:
oh nice, we're already doing the `interface` and `method` naming convention here 😂